components/Form/DynamicStepsInput.js

```diff
@@ -0,0 +1,117 @@
+import styled from "styled-components";
+import { FaRegTrashAlt } from "react-icons/fa";
+import { MdAdd } from "react-icons/md";
+
+export default function DynamicStepsInput({ steps, setSteps }) {
+  function handleAddField() {
+    setSteps([...steps, { id: `${steps.length + 1}`, description: "" }]);
+  }
+
+  function handleRemoveField(indexToRemove) {
+    setSteps(steps.filter((_, index) => index !== indexToRemove));
+  }
+
+  function handleChange(index, event) {
+    const newSteps = [...steps];
+    newSteps[index].description = event.target.value;
+    setSteps(newSteps);
+  }
+
+  return (
+    <>
+      <StyledLabel htmlFor={steps}>Add Steps</StyledLabel>
+      {steps.map((step, index) => (
+        <StyledStepsWrapper key={step.id}>
+          <StyledInput
+            required
+            id={steps}
+            value={step.description}
+            onChange={(event) => handleChange(index, event)}
+            type="text"
+          />
+          <StyledDeleteButton
+            type="button"
+            onClick={() => handleRemoveField(index)}
+          >

```

Comment:
The steps do not always have a consecutive sequence of numbers when steps are deleted during creation

Reply:
Thanks! We would've probably never noticed haha